ipn/ipnlocal/serve.go

```diff
@@ -389,6 +388,24 @@ func (b *LocalBackend) getServeHandler(r *http.Request) (_ ipn.HTTPHandlerView,
 	}
 }
 
+// proxyHandlerForBackend creates a new HTTP reverse proxy for a particular backend that
+// we serve requests for. `backend` is a HTTPHandler.Proxy string (url, hostport or just port).
+func (b *LocalBackend) proxyHandlerForBackend(backend string) (*httputil.ReverseProxy, error) {
+	targetURL, insecure := expandProxyArg(backend)
+	u, err := url.Parse(targetURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid url %s: %w", targetURL, err)
+	}
+	rp := httputil.NewSingleHostReverseProxy(u)
+	rp.Transport = &http.Transport{
```

Comment:
probably more we want to set, like IdleConnTimeout.

See DefaultTransport

Reply:
Thanks, I've copied all values from `DefaultTransport` except for `ProxyFromEnvironment`. Or should we make this respect `HTTP_PROXY` et al?

Reply:
The CLI only allows backends of `localhost/127.0.0.1`. While it's possible to `set-raw` a backend pointing elsewhere, it's not supported. Because of this, I don't think we want to support `HTTP_PROXY`.